pkg/reconciler/taskrun/taskrun.go

```diff
@@ -500,6 +504,23 @@ func (c *Reconciler) reconcile(ctx context.Context, tr *v1beta1.TaskRun, rtr *re
 	return nil
 }
 
+// applyTaskLevelComputeResources applies the task level compute resources to the pod
+func applyTaskLevelComputeResources(tr *v1beta1.TaskRun, rtr *resources.ResolvedTaskResources) {
```

Comment:
this functionality probably belongs in pkg/pod, and I'd prefer to have it in a separate PR because we'll want to make sure it interacts correctly with limitranges.

Reply:
I was thinking to place under `pod`, but thought this logic perhaps only applied on `TaskRun` at this moment? so placing under the taskrun file.

Reply:
but limitranges is a bit complex factor here. was testing around to see how to integrate it into the logic hmm. 👨‍💻

Reply:
It's ok to have slim PRs-- in fact it's usually better :) I do think pkg/pod is the best place and splitting into a separate pr will be better.

Reply:
yap. let me update. think I can include "validation for step-level" in this PR and other logics besides the "applying" related.

thanks Lee.

Reply:
raised https://github.com/tektoncd/pipeline/pull/5082 for "updating" logic